contracts/Vouching.sol

```diff
@@ -1,216 +1,599 @@
+    require(msg.sender != owner(challenge_.entryID), "The owner of a vouched entry can not appeal their own decision");
+
+    Appeal storage appeal_ = challenge_.appeal;
+    appeal_.appealer = msg.sender;
+    appeal_.createdAt = now;
+    appeal_.amount = challenge_.amount.mul(appealFee_).div(PCT_BASE);
+    emit Appealed(_challengeID, msg.sender, appeal_.amount);
+
+    token_.safeTransferFrom(msg.sender, this, appeal_.amount);
+  }
+
+  /**
+   * @dev Accepts an appeal on a challenge. Can only be called by the overseer.
+   */
+  function sustain(uint256 _challengeID) public onlyOverseer existingChallenge(_challengeID) {
+    Challenge storage challenge_ = challenges_[_challengeID];
+    Appeal storage appeal_ = challenge_.appeal;
+    require(challenge_.resolution == Resolution.PENDING, "Given challenge was already resolved");
+    require(appeal_.appealer != address(0), "Cannot sustain a not-appealed challenge");
+
+    challenge_.resolution = Resolution.SUSTAINED;
+    emit Sustained(_challengeID, overseer_);
+
+    uint256 i;
+    uint256 _blocked;
+    Voucher storage voucher_;
+    Entry storage entry_ = entries_[challenge_.entryID];
+
+    if (challenge_.answer == Answer.REJECTED) {
+      entry_.totalVouched = entry_.totalVouched.sub(challenge_.amount);
+
+      for(i = 0; i < entry_.vouchersAddresses.length; i++) {
+        voucher_ = entry_.vouchers[entry_.vouchersAddresses[i]];
+        _blocked = voucher_.blockedPerChallenge[_challengeID];
+        if (_blocked > uint256(0)) voucher_.vouched = voucher_.vouched.sub(_blocked);
+        // no need to reset the voucher's blocked amount for the challenge
+      }
+
+      token_.safeTransfer(appeal_.appealer, appeal_.amount.mul(2));
+      token_.safeTransfer(challenge_.challenger, challenge_.amount.mul(2).sub(appeal_.amount));
+    }
+    else {
+      uint256 profit = challenge_.amount.sub(appeal_.amount);
+      entry_.totalVouched = entry_.totalVouched.add(profit);
+      entry_.totalAvailable = entry_.totalAvailable.add(challenge_.amount).add(profit);
+
+      for(i = 0; i < entry_.vouchersAddresses.length; i++) {
+        voucher_ = entry_.vouchers[entry_.vouchersAddresses[i]];
+        _blocked = voucher_.blockedPerChallenge[_challengeID];
+        if (_blocked > uint256(0)) {
+          uint256 appealPropCost = appeal_.amount.mul(_blocked).div(challenge_.amount);
+          uint256 voucherProfit = _blocked.sub(appealPropCost);
+          voucher_.vouched = voucher_.vouched.add(voucherProfit);
+          voucher_.available = voucher_.available.add(_blocked).add(voucherProfit);
+          // no need to reset the voucher's blocked amount for the challenge
+        }
+      }
+
+      token_.safeTransfer(appeal_.appealer, appeal_.amount.mul(2));
+    }
+  }
+
+  /**
+   * @dev Rejects an appeal on a challenge. Can only be called by the overseer.
+   */
+  function overrule(uint256 _challengeID) public onlyOverseer existingChallenge(_challengeID) {
```

Comment:
I find it odd that there is no code in common between sustain and overrule: I'd expect that affirming a appeal on a rejection should have the same effect than dismissing an appeal on an approval. The only difference should be whether the stake of the appeal is returned to the appealer, or is added to the challenge bounty.

Reply:
Fixed after the [simplification](https://github.com/zeppelinos/zos-vouching/pull/1#issuecomment-458948159) we agreed on the appeals payouts process